include/deal.II/differentiation/ad/adolc_math.h

```diff
@@ -0,0 +1,142 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2016 - 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_differentiation_ad_adolc_math_h
+#define dealii_differentiation_ad_adolc_math_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_ADOLC
+
+DEAL_II_DISABLE_EXTRA_DIAGNOSTICS
+#include <adolc/internal/adolc_settings.h>
+#include <adolc/internal/adubfunc.h> // Taped double math functions
+
+#include <adolc/adouble.h> // Taped double
+#include <adolc/adtl.h>    // Tapeless double
+DEAL_II_ENABLE_EXTRA_DIAGNOSTICS
+
+
+#ifndef DOXYGEN
+
+/**
+ * Import Adol-C math operations into standard namespace. This gives us the
+ * ability to use them within the Tensor class, and it also allows the user
+ * to write generic code and switch between AD number types.
+ *
+ * The math functions to be exposed come from the Adol-C's taped
+ * (anonymous) and tapeless namespaces.
+ */
+namespace std
+{
+
+  /**
+   * Copy a unary function with one name to that of another
```

Comment:
maybe say "make a function with one name available under a different name"? You're copying a "name" here, not a function, after all.